index.js

```diff
@@ -86,6 +85,7 @@ MockDate.prototype.fromLocal = function(d) {
 };
 function localsetter(fn) {
   MockDate.prototype[fn] = function () {
+    if (Number.isNaN(this.d.getTime())) return NaN;
```

Comment:
I think this is wrong, this is for setters, should work even if the date is currently invalid (e.g. `new Date('').setFullYear(2020)` appears to return a valid date).  I think the existing code without the new `if` will work fine (it'l construct a Date with NaN, then call .setUTCFullYear, which does what it needs to).

Reply:
I agree.

Reply:
Updated.